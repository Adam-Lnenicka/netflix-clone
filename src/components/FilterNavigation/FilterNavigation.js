import React from "react";

const FilterNavigation = (prop) => {
  return (
    <>
      <button onClick={() => prop.changeFilter(prop.all)}>All</button>
      <button onClick={() => prop.changeFilter(prop.fantasy)}>Fantasy</button>
      <button onClick={() => prop.changeFilter(prop.drama)}>Drama</button>
      <button onClick={() => prop.changeFilter(prop.crime)}>Crime</button>
      <button onClick={() => prop.changeFilter(prop.action)}>Action</button>
    </>
  );
};

export default FilterNavigation;
