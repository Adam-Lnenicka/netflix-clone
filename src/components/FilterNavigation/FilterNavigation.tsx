import React from "react";
import { filterMovie } from "../../store/actionCreators";
import { useAppDispatch } from "../../store/selectors";

const FilterNavigation = () => {
  const dispatch = useAppDispatch();
  return (
    <div className="filter-navigation">
      <button onClick={() => dispatch(filterMovie("All"))}>All</button>
      <button onClick={() => dispatch(filterMovie("Drama"))}>Drama</button>
      <button onClick={() => dispatch(filterMovie("Fantasy"))}>Fantasy</button>
      <button onClick={() => dispatch(filterMovie("Comedy"))}>Comedy</button>
      <button onClick={() => dispatch(filterMovie("Action"))}>Action</button>
    </div>
  );
};

export default FilterNavigation;
