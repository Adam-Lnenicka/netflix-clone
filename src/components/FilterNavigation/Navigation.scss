.navigation {
  max-width: 1300px;
  padding: 2em 4em;
  padding-bottom: 0;
  margin: 0 auto;
  justify-content: space-between;
}

.filter-navigation {
  border-bottom: 2px #555555 solid;
  button {
    color: white;
    background-color: #232323;
    text-transform: uppercase;
    border: 3px solid #232323;
    cursor: pointer;
    &:hover {
      border-bottom: 3px solid #f65261;
    }
  }
}

.sort-movies {
  background-color: #232323;
  color: white;
  padding: 1rem;
  text-transform: uppercase;
  option {
    background-color: #232323;
    text-transform: uppercase;
  }
}
