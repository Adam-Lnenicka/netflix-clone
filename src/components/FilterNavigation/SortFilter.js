import React from "react";
import { useDispatch } from "react-redux";
import { sortCriteria } from "../../store/actionCreators";

const SortFilter = (prop) => {
  const dispatch = useDispatch();
  return (
    <>
      <span>SORT MOVIES</span>
      <select
        className="sort-movies"
        onChange={(e) => {
          dispatch(sortCriteria(e.target.value));
        }}
      >
        <option>SELECT CRITERIA</option>
        <option value="newest-date">Oldest date</option>
        <option value="oldest-date">Newest Releases</option>
        <option value="title">Title</option>
      </select>
    </>
  );
};

export default SortFilter;
