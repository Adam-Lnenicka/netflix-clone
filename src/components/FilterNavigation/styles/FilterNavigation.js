import styled from "styled-components";

export const FilterNavigation = styled.div`
  .filter-navigation {
    border-bottom: 2px $medium-grey solid;
    button {
      color: white;
      background-color: $light-black;
      text-transform: uppercase;
      border: 3px solid $light-black;
      cursor: pointer;
      &:hover {
        border-bottom: 3px solid $red;
      }
    }
  }
`;
