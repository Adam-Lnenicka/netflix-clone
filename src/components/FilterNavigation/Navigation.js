import React from "react";
import MovieSortCriteria from "../MovieSortCriteria";
import FilterNavigation from "./FilterNavigation";

const Navigation = () => {
  return (
    <div
      className="flex navigation"
      style={{ justifyContent: "space-between" }}
    >
      <FilterNavigation />
      <div className="flex">
        <p>SELECT CRITERIA</p>
        <MovieSortCriteria />
      </div>
    </div>
  );
};

export default Navigation;
