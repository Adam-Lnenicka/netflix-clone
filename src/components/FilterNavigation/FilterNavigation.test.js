import "jsdom-global/register";

import React from "react";
import { mount, shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import FilterNavigation from "./FilterNavigation";

describe("FilterNavigation", () => {
  const dispatch = jest.fn();
  const initialState = {};
  const mockStore = configureStore();
  it("matches snapshot", () => {
    const store = mockStore(initialState);

    const wrapper = shallow(
      <Provider store={store}>
        <FilterNavigation />{" "}
      </Provider>
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("Filter Navigation dispatch triggered on click", () => {
    const store = mockStore(initialState);

    const wrapper = mount(
      <Provider store={store}>
        <FilterNavigation onClick={dispatch} />{" "}
      </Provider>
    );
    wrapper.find("#test-button").simulate("click");
    expect(dispatch).toHaveBeenCalled();
  });
});
