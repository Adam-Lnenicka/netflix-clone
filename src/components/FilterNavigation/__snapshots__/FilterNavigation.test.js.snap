// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterNavigation matches snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <FilterNavigation
    filterMovie={[Function]}
  >
    <div
      className="filter-navigation"
    >
      <button
        id="test-button"
        onClick={[Function]}
      >
        All
      </button>
      <button
        onClick={[Function]}
      >
        Drama
      </button>
      <button
        onClick={[Function]}
      >
        Fantasy
      </button>
      <button
        onClick={[Function]}
      >
        Comedy
      </button>
      <button
        onClick={[Function]}
      >
        Action
      </button>
    </div>
  </FilterNavigation>
   
</Provider>
`;
